lib/pure/strmisc.nim

```diff
@@ -85,6 +86,74 @@ proc rpartition*(s: string, sep: string): (string, string, string)
 
   return partition(s, sep, right = true)
 
+
+func parseFloatThousandSep*(str: openArray[char]; sep = ','; decimalDot = '.'): float {.since: (1, 5).} =
+  ## Convenience func for `parseFloat` which allows for thousand separators,
+  ## this is designed to parse floats as found in the wild formatted for humans.
+  ##
+  ## The following assumptions and requirements must be met:
+  ## - String must not be empty.
+  ## - String must be stripped of trailing and leading whitespaces.
+  ## - `sep` must not be `'-'`.
+  ## - `decimalDot` must not be `'-'` nor `' '`.
+  ## - `sep` and `decimalDot` must be different.
+  ## - Exactly 1 separator must appear after each 3 consecutive inner digits to the left of the dot and nowhere else.
+  ##
+  ## See also:
+  ## * `parseFloat <strutils.html#parseFloat,string>`_
+  runnableExamples:
+    doAssert parseFloatThousandSep("-1,000") == -1000.0
```

Comment:
add negative test to tstrmisc:
parseFloatThousandSep("1,000000")

add positive test to examples:
parseFloatThousandSep("1000,000")
parseFloatThousandSep("1000000")

Reply:
Yes because is the thousand separator no the dot.

Reply:
* what i mean is that `1000,000` (or `1000000,000` ) violates this:

`Exactly 1 separator must appear after each 3 consecutive inner digits to the left of the dot and nowhere else.`

but are accepted by `parseFloatThousandSep`, so that wording needs to be updated.

Reply:
I can revert to the old wording, more verbose but more direct explicit.

Reply:
`parseFloatThousandSep("1000,000")` should certainly not work. Since by default `,` is the thousand separator having it after 4 digits doesn't make sense.